R/bs-theme-preset-bootswatch.R

```diff
@@ -30,8 +27,12 @@ theme_bootswatch <- function(theme) {
 theme_version <- function(theme) {
   if (!is_bs_theme(theme)) return(NULL)
 
-  version <- grep("^bs_version_", class(theme), value = TRUE)
-  sub("^bs_version_", "", version)
+  version_class <- grep("^bs_version_", class(theme), value = TRUE)
+  if (length(version_class) > 0) {
+    sub("^bs_version_", "", version_class)
+  } else {
+    theme_preset_info(theme)$version
+  }
```

Comment:
I'm leaning towards getting rid of this class, but would feel fine keeping it if you feel strongly about it (if you take this suggestion, just make sure to also get rid of the relevant code in `bs_theme_init()`)

```suggestion
  theme_preset_info(theme)$version
```

Reply:
I'm also somewhat ambivalent but lean towards keeping this class. In this case, it's easy to support both and it's faster to check the class. And we currently have the class, so we're not adding something new. I can also see a future in which we might want to use the class to handle the `theme` object differently.

Personally, my inclination would be to use S3 methods in `theme_version()`, where the default uses `theme_preset_info()` but version-specific methods just return the version. 

It turns out that an S3 generic is an order of magnitude faster... but so is directly checking if the theme inherits a known version class

```suggestion
  if (inherits(theme, "bs_version_3")) return("3")
  if (inherits(theme, "bs_version_4")) return("4")
  if (inherits(theme, "bs_version_5")) return("5")
  theme_preset_info(theme)$version
```

<details>
<summary>Reprex Code</summary>

``` r
pkgload::load_all("~/work/rstudio/bslib")
#> ℹ Loading bslib

get_theme_version <- function(theme, ...) {
  UseMethod("get_theme_version", theme)
}

# These methods look better in packaged code, interactively we have to register them
registerS3method(
  "get_theme_version",
  "default",
  \(theme, ...) theme_preset_info(theme)$version
)
registerS3method("get_theme_version", "bs_version_3", \(theme, ...) if (is_bs_theme(theme)) "3")
registerS3method("get_theme_version", "bs_version_4", \(theme, ...) if (is_bs_theme(theme)) "4")
registerS3method("get_theme_version", "bs_version_5", \(theme, ...) if (is_bs_theme(theme)) "5")

# Calling inherits internally turns out to be even faster...
match_theme_version <- function(theme) {
  if (!is_bs_theme(theme)) return(NULL)
  if (inherits(theme, "bs_version_3")) return("3")
  if (inherits(theme, "bs_version_4")) return("4")
  if (inherits(theme, "bs_version_5")) return("5")
  theme_preset_info(theme)$version
}
```

</details>

```r
theme <- bs_theme(version = 5, bootswatch = "flatly")

bench::mark(
  theme_version(theme),      # current version
  get_theme_version(theme),  # S3 dispatch
  match_theme_version(theme) # internal inherits
)
#> # A tibble: 3 × 6
#>   expression                      min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr>                 <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 theme_version(theme)         10.7µs  11.48µs    85040.        0B     25.5
#> 2 get_theme_version(theme)     1.31µs   1.48µs   599057.        0B     59.9
#> 3 match_theme_version(theme) 860.95ns   1.02µs   762283.        0B     76.2
```